src/auth/ejabberd_auth.erl

```diff
@@ -316,6 +328,30 @@ does_user_exist(#jid{luser = LUser, lserver = LServer}) ->
 does_user_exist(error) ->
     false.
 
+%% Hook interface
+-spec does_user_exist(HostType, Jid, RequestType) -> Res when
+      HostType :: mongooseim:host_type(),
+      Jid :: jid:jid(),
+      RequestType :: stored | with_anonymous,
```

Comment:
Why not use `exist_type`? I also think that the variable names don't give any extra information in this case, only more lines.